src/app/contacts.service.ts

```diff
@@ -0,0 +1,58 @@
+import { Injectable } from '@angular/core'
+import { CONTACTS, Contact } from './contacts/contacts'
+import { last } from 'rxjs'
+
+@Injectable({
+  providedIn: 'root',
+})
+export class ContactsService {
+  public contacts: Contact[] = CONTACTS
+
+  getContact(contactId: number): Contact | null {
+    const foundContact = this.contacts.find((c) => c.id === contactId)
+    return foundContact || null
+  }
+
+  addContact(
+    firstName: string,
+    lastName: string,
+    street: string,
+    city: string
+  ): void {
+    const newContact: Contact = {
+      id:
+        this.contacts.length > 0
+          ? this.contacts[this.contacts.length - 1].id + 1
```

Comment:
For generating new contact IDs, relying on the last contact's ID might not be ideal, especially if you plan to allow deleting contacts in the future. Consider using a more robust strategy for generating unique identifiers, such as UUIDs, to avoid potential ID conflicts